dali/test/python/test_operator_caffe_reader.py

```diff
@@ -0,0 +1,201 @@
+
+        self.decode = ops.ImageDecoderCrop(device = "cpu",
+                                           crop = (224, 224),
+                                           crop_pos_x = 0.3,
+                                           crop_pos_y = 0.2,
+                                           output_type = types.RGB)
+    def define_graph(self):
+        inputs, labels = self.input(name="Reader")
+
+        images = self.decode(inputs)
+        return images, labels
+
+# test 1: rewind
+def check_reader_vs_rewind(path):
+    # the db has 43 entries, 44 should trigger rewind
+    if isinstance(path, str):
+        num_paths = 1
+    else:
+        num_paths = len(path)
+    num = 43*num_paths + 1
+    pipe = CaffeReaderPipeline(path, 1)
+    pipe.build()
+    for i in range(num):
+        pipe_out = pipe.run()
+        if i == 0:
+            image0 = pipe_out[0].at(0)
+            label0 = pipe_out[1].at(0)
+        if i == num-1:
+            image0_rewind = pipe_out[0].at(0)
+            label0_rewind = pipe_out[1].at(0)
+        
+    assert_array_equal(image0, image0_rewind)
+    assert_array_equal(label0, label0_rewind)
+
+def test_reader_vs_rewind():
+    for path in [caffe_db_folder, [caffe_db_folder], [caffe_db_folder, caffe_db_folder]]:
+        yield check_reader_vs_rewind, path
+
+# test 2: compare with previous CaffeReader
+# run the previous CaffeReader to get the db statistics [image.mean(), label], 
+# regarded as ground truth
+db_stat = [
+    [118.10, 1],
+    [95.21, 1],
+    [105.89, 1],
+    [135.78, 1],
+    [152.72, 1],
+    [93.08, 0],
+    [153.97, 0],
+    [177.89, 0],
+    [163.62, 1],
+    [82.04, 1],
+    [214.37, 1],
+    [77.89, 1],
+    [231.43, 0],
+    [100.98, 1],
+    [157.01, 1],
+    [167.31, 1],
+    [98.75, 0],
+    [101.37, 1],
+    [125.64, 1],
+    [120.80, 1],
+    [128.53, 1],
+    [95.86, 1],
+    [56.78, 0],
+    [89.86, 1],
+    [86.71, 1],
+    [118.33, 1],
+    [76.55, 1],
+    [108.08, 0],
+    [147.95, 0],
+    [188.69, 0],
+    [135.47, 1],
+    [164.58, 0],
+    [172.21, 1],
+    [111.00, 1],
+    [175.26, 0],
+    [99.32, 1],
+    [138.10, 1],
+    [97.69, 1],
+    [108.19, 0],
+    [60.03, 0],
+    [145.58, 1],
+    [135.89, 1],
+    [141.06, 0],
+]
+
+def check_reader_vs_db_stat(path, batch_size, num_threads, num_gpus):
+    pipelines = [CaffeReaderPipeline(path, batch_size, num_threads, device_id, num_gpus) for device_id in range(num_gpus)]
+    
+    num_batches = 2
+    for pipe in pipelines:
+        pipe.build()
+        count = 0
+        for i in range(num_batches):
+            pipe_out = pipe.run()
+            for idx in range(len(pipe_out[0])):
+                image = pipe_out[0].at(idx)
+                label = pipe_out[1].at(idx)
+                dstid = count % len(db_stat)
+                assert_allclose(image.mean(), db_stat[dstid][0], rtol=1e-3)
+                assert_array_equal(label[0], db_stat[dstid][1])
+                count+=1
+
+def test_reader_vs_db_stat():
```

Comment:
I don't know if checking `caffe_db_folder` against the ground truth is not too fragile. Maybe it is enough to online generate the ground truth for `caffe_db_folder` and test it against `[caffe_db_folder], [caffe_db_folder, caffe_db_folder]` later.

Reply:
Thank you for your review. 
Do you mean creating an instance of `CaffeReader(path = caffe_db_folder)` and then testing it against the instance of `CaffeReader(path = [caffe_db_folder])` and `CaffeReader(path = [caffe_db_folder, caffe_db_folder])`?

Reply:
Yes, that is the idea. But it is not a must, rather an idea.

Reply:
Thank you for your reply. I removed the fragile tests that depend on a specific LMDB file, and added the test as you said.